models/user/user.go

```diff
@@ -1219,6 +1227,43 @@ func GetAdminUser() (*User, error) {
 	return &admin, nil
 }
 
+func isUserVisibleToViewerCond(viewer *User) builder.Cond {
+	cond := builder.NewCond()
+	if viewer != nil && viewer.IsAdmin {
+		return cond
+	}
+	cond = builder.Eq{
+		"`user`.Visibility": structs.VisibleTypePublic,
+	}
+
+	if viewer == nil || viewer.IsRestricted {
+		return cond
+	}
+
+	cond = builder.Not{builder.Eq{
```

Comment:
We have builder.Neq and we also why are we not considering visibility private?

Reply:
I tried NEq not Neq...

Look again at the code regarding your second question.